src/senselab/utils/device.py

```diff
@@ -0,0 +1,35 @@
+"""Utility functions for utilizing different devices in Senselab."""
+
+from enum import Enum
+
+import torch
+
+
+class DeviceType(Enum):
+    """Device types for PyTorch operations."""
+
+    CPU: str = "cpu"
+    CUDA: str = "cuda"
+    MPS: str = "mps"
+
+
+def _select_device_and_dtype(
```

Comment:
should we maybe have 3 params?
- requested_device (optional)
- available_devices (we can compute this)
- compatible_devices (this depends on the models)

in this way, we can more easily handle when the user asks for a device that is not available or compatible